lib/srv/server/installer/defaultinstallers.go

```diff
@@ -26,19 +26,48 @@ import (
 	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
 )
 
-// DefaultInstaller represents the default installer script provided by teleport.
-var DefaultInstaller = oneoffScriptToDefaultInstaller()
+// LegacyDefaultInstaller represents the default installer script provided by teleport.
+var (
+	// LegacyDefaultInstaller uses oneoff.sh to download the Teleport tarball and run `teleport install`.
+	// The Teleport install command handles both Teleport installation and agent configuration.
+	LegacyDefaultInstaller = oneoffScriptToDefaultInstaller()
 
-func oneoffScriptToDefaultInstaller() *types.InstallerV1 {
-	argsList := []string{
+	// NewDefaultInstaller installs Teleport by calling the standard "/scripts/install.sh" route on the proxy.
+	// After successfully installing Teleport, it will invoke the same `teleport install`
+	// command as the LegacyDefaultInstaller which will only take care of configuring Teleport.
+	NewDefaultInstaller = types.MustNewInstallerV1(
+		installers.InstallerScriptName,
+		strings.Join(
+			[]string{scriptShebangAndSetOptions, execGenericInstallScript, configureTeleport},
+			"\n\n",
+		),
+	)
+
+	scriptShebangAndSetOptions = `#!/bin/bash
+set -euo pipefail`
+	execGenericInstallScript = `
+INSTALL_SCRIPT_URL="https://{{.PublicProxyAddr}}/scripts/install.sh"
+
+echo "Offloading the installation part to the generic Teleport install script hosted at: $INSTALL_SCRIPT_URL"
+
+curl "$INSTALL_SCRIPT_URL" | sudo bash`
```

Comment:
Shouldn't we exit if the script fails?

Also, `curl -sSf`?

Reply:
AFAIK set -e should take care of this (and -o pipefail ensures that we also stop if curl fails) but I can do a nice error handling:

```suggestion
curl "$INSTALL_SCRIPT_URL" | sudo bash || (echo "Failed to execute the install.sh script" && exit 1)`
```